From 42c25efeaa409f711af1bd9a8144b445741ee114 Mon Sep 17 00:00:00 2001
From: Juha Vuolle <juha.vuolle@insta.fi>
Date: Mon, 14 Mar 2022 13:03:27 +0200
Subject: [PATCH 5/5] Add QIODevice::canReadLine call to QBluetoothSocket

At least on Linux the QIODevice parent class reads and buffers a
large chunk of data when readLine() is called. This call effectively
empties the private linear data buffer, and subsequent canReadLine()
calls to the private buffer returns false. This is problematic if
the data contained several lines of data; the
QBluetoothSocket::canReadLine() returns false but readLine() returns
valid lines of data.

This commit adds a parent class call so that it's buffers are also
taken into account as per QIODevice documentation.

Fixes: QTBUG-101690
Pick-to: 5.15 6.2 6.3
Change-Id: I8130aff217e9e6c5525101901ed55721430b6dd0
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
Reviewed-by: Alex Blasche <alexander.blasche@qt.io>
(cherry picked from commit 393fd47ab64ce80fb0e852027591840aef8e135d)
---
 src/bluetooth/qbluetoothsocket.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bluetooth/qbluetoothsocket.cpp b/src/bluetooth/qbluetoothsocket.cpp
index e4d85447..aadd4755 100644
--- a/src/bluetooth/qbluetoothsocket.cpp
+++ b/src/bluetooth/qbluetoothsocket.cpp
@@ -593,7 +593,7 @@ void QBluetoothSocket::setSocketState(QBluetoothSocket::SocketState state)
 bool QBluetoothSocket::canReadLine() const
 {
     Q_D(const QBluetoothSocketBase);
-    return d->canReadLine();
+    return d->canReadLine() || QIODevice::canReadLine();
 }
 
 /*!
-- 
2.36.1

